cmd/metrics-v3-cache.go

```diff
@@ -108,14 +131,35 @@ func newDriveMetricsCache() *cachevalue.Cache[storageMetrics] {
 		storageInfo := objLayer.LocalStorageInfo(GlobalContext, true)
 		onlineDrives, offlineDrives := getOnlineOfflineDisksStats(storageInfo.Disks)
 		totalDrives := onlineDrives.Merge(offlineDrives)
+
+		var ioStats *diskIOStats
+		currentStats := getCurrentDiskMetrics()
+		if initialDriveIOStats == nil {
+			initialDriveIOStats = &diskIOStats{
+				stats:       currentStats,
+				collectedAt: time.Now().UTC(),
+			}
+		} else {
+			diffStats := map[string]madmin.DiskIOStats{}
+			for d, cs := range currentStats {
+				if is, found := initialDriveIOStats.stats[d]; found {
+					diffStats[d] = getDiffStats(is, cs)
+				}
+			}
+			ioStats = &diskIOStats{stats: diffStats, collectedAt: time.Now().UTC()}
```

Comment:
It seems like you are missing the duration of the collected data, otherwise the metrics cannot be calculated as a rate over time.

Edit: It seems like you are hacking that in below: `durationSecs := ioStats.collectedAt.Sub(initialDriveIOStats.collectedAt).Seconds()` - do it here.

Reply:
done, capturing `duration` instead of `collectedAt` now.